src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+    try (BufferedWriter bw = new BufferedWriter(new FileWriter(newFileName))) {
+      bw.write(compUnit.toString());
+    } catch (IOException e) {
+      System.err.println("Error writing to file: " + newFileName);
+      e.printStackTrace();
+      System.exit(1);
+    }
+
+    return newFileName;
+  }
+
+  /**
+   * Given the absolute file path to a Java file, return the simple class name of the file.
+   *
+   * @param filePath absolute path to a Java file
+   * @return the simple class name
+   */
+  public static String getClassName(String filePath) {
+    if (filePath == null) {
+      return null;
+    }
+
+    // Get the name of the file without full path
+    String fileName = Paths.get(filePath).getFileName().toString();
+    // Remove .java extension from file name
+    return fileName.substring(0, fileName.indexOf('.'));
+  }
+
+  /**
+   * Rename the overall class. Visit every class and interface declaration, of which we expect there
+   * to be one in the input test suite. The overall class will be renamed to class name + suffix.
+   */
+  private static class ClassRenamer extends VoidVisitorAdapter<Object> {
+    /**
+     * @param arg a String array where the first element is the class name and the second element is
+     *     the suffix that we will append
+     */
+    @Override
+    public void visit(ClassOrInterfaceDeclaration n, Object arg) {
+      String[] params = (String[]) arg;
+      String className = params[0];
+      String suffix = params[1];
+      if (className.equals(n.getName())) {
+        n.setName(className + suffix);
+      }
+    }
+  }
+
+  private static class TypeVisitor extends VoidVisitorAdapter<Object> {
+    /** @param arg a set of {@code Type} objects */
+    @SuppressWarnings("unchecked")
+    @Override
+    public void visit(ClassOrInterfaceType n, Object arg) {
+      Set<Type> params = (Set<Type>) arg;
+      if (n.toString().contains(".")) {
+        params.add(n);
+      }
+    }
+  }
+
+  /**
+   * Add an import to the list of {@code ImportDeclaration}s of the compilation unit. This method
+   * adds the import statement if it has not been yet included in the list of current imports of the
+   * compilation unit.
+   *
+   * @param compilationUnit compilation unit to add import to
+   * @param name the import name
+   */
+  private static void addImport(CompilationUnit compilationUnit, String name) {
+    String i = "import " + name + ";";
+    ImportDeclaration importDeclaration;
+
+    try {
+      importDeclaration = JavaParser.parseImport(i);
+    } catch (ParseException e) {
+      // Unexpected error from parsing import.
+      System.err.println("Error parsing import");
+      return;
+    }
+
+    List<ImportDeclaration> importDeclarations = compilationUnit.getImports();
+    for (ImportDeclaration im : importDeclarations) {
+      // Check if the compilation unit already includes the import.
+      if (importDeclaration.toString().equals(im.toString())) {
```

Comment:
It would be good to check for wildcards too.  Please add a test case that contains
  `import java.util.*;`
and make sure that you don't add a redundant
  `import java.util.List;`